packages/input-mask/README.md

```diff
@@ -0,0 +1,82 @@
+# @solid-primitives/input-mask
+
+[![lerna](https://img.shields.io/badge/maintained%20with-lerna-cc00ff.svg?style=for-the-badge)](https://lerna.js.org/)
+[![size](https://img.shields.io/bundlephobia/minzip/@solid-primitives/input-mask?style=for-the-badge&label=size)](https://bundlephobia.com/package/@solid-primitives/input-mask)
+[![version](https://img.shields.io/npm/v/@solid-primitives/input-mask?style=for-the-badge)](https://www.npmjs.com/package/@solid-primitives/input-mask)
+[![stage](https://img.shields.io/endpoint?style=for-the-badge&url=https%3A%2F%2Fraw.githubusercontent.com%2Fdavedbase%2Fsolid-primitives%2Fmain%2Fassets%2Fbadges%2Fstage-0.json)](https://github.com/davedbase/solid-primitives#contribution-process)
+
+Primitive that returns an event handler to mask the inputs of an text input element (`<input>`, `<textarea>`) when applied in `oninput` or `onchange`.
+
+## Installation
+
+```bash
+npm install @solid-primitives/input-mask
+# or
+yarn add @solid-primitives/input-mask
+```
+## Usage
+
+The masks come in 3 different formats: function, array and string. There are tools to convert string masks to array masks and array masks to function masks.
+
+```ts
+// string mask:
+// 9 = any number,
+// a = any letter,
+// * = any alphanumeric character
+// any other letter becomes a fixed placeholder
+const isodate = '9999-99-99';
+// array mask: RegExp to match variable parts, strings for fixed placeholders
+const meetingId = [/\d/, /\d/, /\d/, '-', /\d/, /\d/, /\d/, '-', /\d/, /\d/, /\d/];
+// function mask: (value, [start, end]) => [value, [start, end]]
+const meetingIdOrName = (value, selection) => (/^\D/.test(value)
+  ? arrayMaskToFn([/\w+/, /\S*/])
+  : anyMaskToFn('999-999-999'))(value, selection);
+
+// converting string mask to array:
+const maskArrayFromString = stringMaskToArray(maskString);
+const maskFuncFromArray = arrayMaskToFn(maskArray);
+const maskFuncFromStringOrArray = anyMaskToFn(mask);
+
+return <>
+  <label for="changeMask">
+    The mask will only be applied after you leave the field
+  </label>
+  <input type="text" id="changeMask" onchange={createInputMask('99/99/9999')} />
+  <label for="inputMask">
+    The mask will be applied on every single input
+  </label>
+  <input type="text" id="inputMask" oninput={createInputMask('99/99/9999')} />
```

Comment:
How would you actually get a value out of the input? I think it's worth to include an example here, where you save the masked input value to some signal.

Reply:
Good idea. I added two ways. I'm still considering if I should make the handler return the value for convenience reasons. It wouldn't exactly fit the signature of an event handler anymore if typed strictly to emit void, but this would allow to use the return value.

Reply:
So the approach with "handler returns the value" would look like this?
```tsx
const [date, setDate] = createSignal('01/01/2020');
const mask = createInputMask('99/99/9999');
return <input onInput={e => setDate(mask(e))} />
```
This seams pretty clean, but how would the other approach look like?

Reply:
Oh, and I guess there is a problem with the initial value, you cannot use signal for that...
```tsx
const initial = "01/01/2020";
const [date, setDate] = createSignal(initial);
const mask = createInputMask("99/99/9999");
return <input value={initial} onInput={e => setDate(mask(e))} />
```

Reply:
Also how setting the value from outside would look like?
For example if you have a date picker with masked text input and calendar component connected to each other.
How would you apply value to the input, when user changes a day using the calendar?

Reply:
Just set the value of the input. Overwriting the value from JS is not going to trigger the handler.

Reply:
You also could still use it to write the let variable:

```ts
let date = "01/01/2020";
const mask = createInputMask("99/99/9999");
return <input value={date} onInput={e => { date = mask(e) }} />
```

Reply:
Yes, you've convinced me to add this.

Reply:
So something like this? If this works, then that's great.
```ts
const [date, setDate] = createSignal("01/01/2020");
const handler = createInputMask("99/99/9999");
return <input value={date()} onInput={e => setDate(handler(e))} />

// later, a change from outside
setDate("12/09/2021")
```